2007-10-18  Ulrich Drepper  <drepper@redhat.com>

	* sunrpc/clnt_udp.c (clntudp_call): Don't block in recvfrom call
	even if the poll result indicates there is data to read.
	Patch by Jeff Moyer <jmoyer@redhat.com>.

--- libc/sunrpc/clnt_udp.c	20 Dec 2005 22:38:40 -0000	1.33
+++ libc/sunrpc/clnt_udp.c	18 Oct 2007 22:23:57 -0000	1.34
@@ -413,7 +413,7 @@ send_again:
 	{
 	  fromlen = sizeof (struct sockaddr);
 	  inlen = __recvfrom (cu->cu_sock, cu->cu_inbuf,
-			      (int) cu->cu_recvsz, 0,
+			      (int) cu->cu_recvsz, MSG_DONTWAIT,
 			      (struct sockaddr *) &from, &fromlen);
 	}
       while (inlen < 0 && errno == EINTR);
